app.py

```diff
@@ -102,357 +125,270 @@ def toggle_left_column(left_column_collapse: int, class_name: str) -> str:
-        dict: Style for the D-Wave summary table
-        dict: Style for the Classical summary table
+        str: Run button class.
+        str: Cancel button class.
     """
-    if ctx.triggered_id == "run-button":
-        if run_click == 0:
-            empty_figure = get_empty_figure("Run optimization to see results")
-            return (
-                "tab",
-                "tab",
-                empty_figure,
-                empty_figure,
-                {"visibility": "hidden"},
-                {"visibility": "hidden"},
-            )
-        else:
-            empty_figure = get_empty_figure("Running...")
-            return (
-                "tab-loading",
-                "tab-loading",
-                empty_figure,
-                empty_figure,
-                {"visibility": "hidden"},
-                {"visibility": "hidden"},
-            )
-    elif ctx.triggered_id == "cancel-button":
-        if cancel_click > 0:
-            empty_figure = get_empty_figure(
-                "Last run cancelled prior to completion. Re-run to see results"
-            )
-            return (
-                "tab",
-                "tab",
-                empty_figure,
-                empty_figure,
-                {"visibility": "hidden"},
-                {"visibility": "hidden"},
-            )
-    raise PreventUpdate
+    if not running_classical and not running_dwave:
+        return "", "display-none"
+
+    return "display-none", ""
 
 
 @app.callback(
-    Output("optimized_gantt_chart", "figure"),
-    Output("dwave_summary_table", "figure"),
-    Output("dwave_tab", "className"),
-    Output("dwave_summary_table", "style"),
+    Output("optimized-gantt-chart", "figure"),
+    Output("dwave-summary-table", "figure"),
+    Output("dwave-tab", "className"),
+    Output("dwave-tab", "label"),
+    Output("dwave-tab", "disabled"),
+    Output("running-dwave", "data"),
     background=True,
     inputs=[
         Input("run-button", "n_clicks"),
         State("model-select", "value"),
         State("solver-select", "value"),
         State("scenario-select", "value"),
-        State("solver_time_limit", "value"),
-    ],
-    running=[
-        (Output("cancel-button", "style"), {"display": "inline-block"}, {"display": "none"}),
-        (Output("run-button", "style"), {"display": "none"}, {"display": "inline-block"}),
+        State("solver-time-limit", "value"),
     ],
     cancel=[Input("cancel-button", "n_clicks")],
     prevent_initial_call=True,
 )
 def run_optimization_cqm(
-    run_click: int, model: str, solver: str, scenario: str, time_limit: int
-) -> tuple[go.Figure, go.Figure, str, dict]:
+    run_click: int, model: int, solvers: list[int], scenario: str, time_limit: int
+) -> tuple[go.Figure, go.Figure, str, str, bool, bool]:
     """Runs optimization using the D-Wave hybrid solver.
 
     Args:
         run_click (int): The number of times the run button has been
             clicked.
```

Comment:
This actually fits in the 100 char line limit.
```suggestion
        run_click (int): The number of times the run button has been clicked.
```